pkg/manager/manager.go

```diff
@@ -120,7 +123,7 @@ func (k *KubeMacPoolManager) Run(rangeStart, rangeEnd net.HardwareAddr) error {
 			return errors.Wrap(err, "failed to start pool manager routines")
 		}
 
-		err = k.runtimeManager.Start(context.Background())
+		err = k.runtimeManager.Start(ctx)
```

Comment:
is it fine that there is no return on line 128  (right column of the numbers) ?

general speaking - maybe better to explicitly close channels for example on line 132
when creating a new one

Reply:
when KMP runtimeManager fails, it does a "soft" restart, where we simply return to the runtimeManager loop. This is why if `k.runtimeManager.Start(ctx)` returns, we don't return and simply reloop (if it would return, the the main function will finish, and the pod will be restarted by the deployment, which is more like a "hard" reset.

As to explicitly close channels.. I was under the impression that closing channels is the way channels signal. so not sure closing them is the right way to handle them.. @qinqon is there a better way to handle this?